### ISSUE TITLE
Custom header and footer partials

### ISSUE THREAD

**Issue by ColtonProvias** [solutionroute/hugo-smorg#4]
To make it easier for users to add scripts/stylesheets/meta tags/etc., it may be ideal to use custom header and footer partials. You can see this implemented in the [Learn theme](https://github.com/matcornic/hugo-theme-learn/tree/master/layouts/partials).

**Comment by solutionroute** [comment]
Just to confirm - we'll use a partial named custom-PURPOSE (header / footer / etc) - provide a default implementation in [theme]/layout/partials which any site can override in their site/layouts/partials?

Great idea... I like that much better than an overly complicated config file. +1, coming soon.

**Comment by ColtonProvias** [comment]
Yes, that's the idea. It made it easy to add on KaTeX and other scripts for processing while still keeping the theme updated.

**Comment by solutionroute** [comment]
I messed up closing with the actual commit that did the work but you'll see it easily enough. I may have gone farther than I should with the changes; any that prove superfluous in the longer run will eventually disappear. I think that I captured the spirit of what you were looking for.
